### DISCUSSION TITLE
How can I add plugins to eslint config ?

### DISCUSSION THREAD

**Discussion by p-l-l** [teambit/bit#4914] [Help]
Hi, I am trying to add the react-hooks plugin to my custom env eslint config.
I did use Bit install to have the following depency "eslint-plugin-react-hooks": "4.2.0"

I am using this code to configure my linting rules in a custom env:
```
react.useEslint({
        transformers: [
          (config) => {
            config.addPlugins(['react-hooks']);
            config.setRule("react-hooks/rules-of-hooks", "error");
            config.setRule("react-hooks/exhaustive-deps", "warn");
           
            return config;
          },
        ],
      }),
```
Everything's been working fine (with other rules) until I used the `config.addPlugins(['react-hooks']);` line, which throws the
following error when using `bit lint` :
```
service linter env pll-org.design/extensions/custom-react has failed. Invalid Options:
- Unknown options: extends
```

Also made sure to `bit build` -> `bit compile` -> `bit lint` with the saved changes.

Is there any example on how a plugin can be added ? Am I missing a step ?

**Comment by GiladShoham** [comment]
It might be a bug. I'll check it with in a few days and will let you know

**Reply by p-l-l** [reply]
Hi @GiladShoham, been taking a break from my Bit project for a while, now I am back at it. I saw this interesting way of sharing eslint config from [learn-bit-react](https://bit.dev/learn-bit-react/configs/my-eslint) and I made a config and was able to make it work in VS Code.
I am trying to reuse it in my custom env extension
```
react.useEslint({
  transformers: [
	(config) => {
		config.raw.config.baseConfig = eslintConfig;
		return config;
	},
  ],
}),
```

I still have trouble when it comes to extends and plugins
![image](https://user-images.githubusercontent.com/23360637/155232075-d11271f3-17f1-4806-bd1a-b15e2cac4d42.png)

How do I provide the dependency (in this case `eslint-config-react-app`) ? I tried adding it to `overrideDependencies` as devDependencies, but still got the same error..

Thanks for any help!

**Reply by benjgil** [reply]
@p-l-l 
I can't say whether this will solve your issue, but when it comes to your env the eslint packages are regular deps, not devDependencies - they are used in the core of the component, not just to develop it.
It's part of the mental shift to component-driven development. Once you can have components which themselves define the dev process, you then have situations where packages which are usually devDeps become regular dependencies.

So tl;dr - try adding them as regular dependencies
